ttnn/cpp/ttnn/operations/sharding_utilities.cpp

```diff
@@ -751,27 +751,132 @@ ShardingConfig get_specs_for_sharding_partition(
+        size_t offset = 1 + !rank_crta + !num_banks_crta + rank * !tensor_shape_crta;
+        return tt::stl::Span<const uint32_t>(compile_time_args.data() + offset, rank);
+    }
+}
+tt::stl::Span<const uint32_t> ShardedAccessorArgs::get_bank_coords() const {
+    // if bank_coords is runtime time
+    auto rank = get_rank();
+    bool rank_crta = args_config.test(ArgConfig::RankCRTA);
+    bool num_banks_crta = args_config.test(ArgConfig::NumBanksCRTA);
+    bool tensor_shape_crta = args_config.test(ArgConfig::TensorShapeCRTA);
+    bool shard_shape_crta = args_config.test(ArgConfig::ShardShapeCRTA);
+    if (args_config.test(ArgConfig::BankCoordsCRTA)) {
+        size_t offset = rank_crta + num_banks_crta + rank * (tensor_shape_crta + shard_shape_crta);
+        return tt::stl::Span<const uint32_t>(runtime_args.data() + offset, get_physical_num_banks());
+    } else {
+        size_t offset = 1 + !rank_crta + !num_banks_crta + rank * (!tensor_shape_crta + !shard_shape_crta);
+        return tt::stl::Span<const uint32_t>(compile_time_args.data() + offset, get_physical_num_banks());
+    }
+}
+
 ShardedAccessorArgs get_sharded_accessor_args(
     const distributed::MeshDevice& mesh_device,
     const BufferDistributionSpec& buffer_distribution_spec,
-    const CoreType& bank_type) {
+    const CoreType& bank_type,
+    const ArgsConfig& args_config) {
     const auto& tensor_shape = buffer_distribution_spec.get_tensor_shape_in_pages();
     const auto& shard_shape = buffer_distribution_spec.get_shard_shape_in_pages();
     const auto& bank_coords = buffer_distribution_spec.get_cores();
 
-    std::vector<uint32_t> shapes_and_bank_coords;
-    shapes_and_bank_coords.reserve(tensor_shape.size() + shard_shape.size() + bank_coords.size());
-    shapes_and_bank_coords.insert(shapes_and_bank_coords.end(), tensor_shape.cbegin(), tensor_shape.cend());
-    shapes_and_bank_coords.insert(shapes_and_bank_coords.end(), shard_shape.cbegin(), shard_shape.cend());
-
-    // Pack each virtual coordinate as a 32-bit value with 16 bits for x and 16 bits for y
-    for (const auto& bank_coord : bank_coords) {
-        const auto virtual_coord = mesh_device.virtual_core_from_logical_core(bank_coord, bank_type);
-        shapes_and_bank_coords.push_back((virtual_coord.x << 16) | (virtual_coord.y & 0xFFFF));
+    auto rank_rt = args_config.test(ArgConfig::RankCRTA);
+    auto num_banks_rt = args_config.test(ArgConfig::NumBanksCRTA);
+    auto tensor_shape_rt = args_config.test(ArgConfig::TensorShapeCRTA);
+    auto shard_shape_rt = args_config.test(ArgConfig::ShardShapeCRTA);
+    auto bank_coords_rt = args_config.test(ArgConfig::BankCoordsCRTA);
+
+    TT_FATAL(
+        !rank_rt or (rank_rt and tensor_shape_rt and shard_shape_rt),
+        "If rank is runtime, tensor_shape and shard_shape must also be runtime");
+    TT_FATAL(
+        !num_banks_rt or (num_banks_rt and bank_coords_rt),
+        "If num_banks is runtime, bank_coords must also be runtime");
```

Comment:
I think and, or should be &&, || for consistent in codebase.
```suggestion
    TT_FATAL(!rank_rt || (tensor_shape_rt && shard_shape_rt),
             "If rank is runtime, tensor_shape and shard_shape must also be runtime");
    TT_FATAL(!num_banks_rt || bank_coords_rt,
             "If num_banks is runtime, bank_coords must also be runtime");
```